go/libraries/doltcore/sqle/sglobal/auto_increment_tracker.go

```diff
@@ -0,0 +1,135 @@
+package sglobal
+
+import (
+	"fmt"
+	"sync"
+)
+
+type AutoIncrementTracker interface {
```

Comment:
This probably should go in the sqle package, no reason to put it in the global state package

Reply:
I honestly prefer it here, because it signals better its intended use.